src/readers/stream.ts

```diff
@@ -19,15 +19,27 @@ export default class ReaderStream extends Reader<NodeJS.ReadableStream> {
 		const filepaths = patterns.map(this._getFullEntryPath, this);
 
 		const stream = new PassThrough({ objectMode: true });
+		let matches = 0;
 
 		stream._write = (index: number, _enc, done) => {
+			if (options.maxMatches === matches) {
+				// This is not ideal because we are still passing patterns to write
```

Comment:
Help would be appreciated here @mrmlnc - I think the current solution is okayish, however ideally we would backpressure the stream in `for (let i = 0; i < filepaths.length; i++) {` based on the async `stream.write` task. Happy to follow another route as well if you have ideas.